### ISSUE TITLE
[routing] preg_match(): Compilation failed: unmatched parentheses

### ISSUE THREAD

**Issue by zajca** [symfony/symfony#27504]
**Symfony version(s) affected**: 4.1.0

**Description**  
I'm having problem with router after 4.1 update on one of my routes I'm getting: `Uncaught PHP Exception ErrorException: "Warning: preg_match(): Compilation failed: unmatched parentheses at offset 595" at /var/www/symfony/var/cache/dev/srcDevDebugProjectContainerUrlMatcher.php line 415` This looks like there is something wrong with generated regex.

**Additional context**  

Here is generated regex
https://pastebin.com/egUBavqi

Routes in switch that matches url
```
849 => array(array('_route' => 'nsp_admin_package_createaction__invoke', '_format' => 'json', '_controller' => 'NSP\\Controller\\Admin\\Package\\CreateAction'), array('_format'), array('POST' => 0), null),
978 => array(array('_route' => 'nsp_admin_package_listaction__invoke', '_format' => 'json', '_controller' => 'NSP\\Controller\\Admin\\Package\\ListAction'), array('_format'), array('GET' => 0), null),
```

I'm calling: `GET /admin/api/package.json`

**Comment by nicolas-grekas** [comment]
Can you provide a small reproducer? E.g. a small set of routes that reproduce this?

**Comment by nicolas-grekas** [comment]
The regexps you linked on pastbin are valild, they do not generate this error.

Status: waiting feedback

**Comment by zajca** [comment]
I spend some time with this, removing vendor and do clean install helps :/ probably something wrong with update, sorry to bother.

**Comment by zajca** [comment]
Sorry again, I just try run project in CI in clean environment and error still persists.
I'm not able to create minimal example. But figure out that it should be possible to run whole srcDevDebugProjectContainerUrlMatcher.php class alone.
So i created reproducer her: https://github.com/zajca/sf41-bug-routing-27504
There is whole auto-generated class with `test.php` which will trigger error.

**Comment by nicolas-grekas** [comment]
I'm sorry but I cannot use your reproducer as the compiled container is just unreadable.
Anyway, can you check if #27511 fixes the issue?
If not, please provide a reproducer that starts from regular route configuration, so that I can compile on my own.
Thanks in advance.

**Comment by zajca** [comment]
I've checked #27511 but srcDevDebugProjectContainerUrlMatcher is just same.
I also tried to make reproducer by route configuration but routes are just fine, there is nothing special. Also all regex in srcDevDebugProjectContainerUrlMatcher are fine but still trigger error.

**Comment by zajca** [comment]
OK, i did reproduce this with 4 routes:
https://github.com/zajca/sf41-bug-routing-27504

**Comment by nicolas-grekas** [comment]
Will be fixed by #27511
